src/cursor/cur_std.c

```diff
@@ -1113,6 +1113,55 @@ __wt_cursor_reconfigure(WT_CURSOR *cursor, const char *config)
     API_END_RET(session, ret);
 }
 
+/*
+ * __wt_cursor_largest_key --
+ *     WT_CURSOR->largest_key default implementation..
+ */
+int
+__wt_cursor_largest_key(WT_CURSOR *cursor)
+{
+    WT_DECL_ITEM(key);
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+    WT_TXN *txn;
+    bool ignore_tombstone;
+
+    txn = NULL;
+    ignore_tombstone = F_ISSET(cursor, WT_CURSTD_IGNORE_TOMBSTONE);
+    CURSOR_API_CALL(cursor, session, largest_key, NULL);
+
+    txn = session->txn;
+    if (F_ISSET(txn, WT_TXN_SHARED_TS_READ))
+        WT_ERR_MSG(session, EINVAL, "largest key cannot be called with a read timestamp");
+
+    WT_ERR(__wt_scr_alloc(session, 0, &key));
+
+    /* Reset the cursor to give up the cursor position. */
+    WT_ERR(cursor->reset(cursor));
```

Comment:
can we restore the cursor state in case of an error by using the pair "__cursor_state_save" and "__cursor_state_restore"

Reply:
This is only used at btree cursor level. I don't think we should do it here.

Reply:
Why we need to do this?